search/views.py

```diff
@@ -0,0 +1,70 @@
+""" handle requests for courseware search http requests """
+import logging
+import json
+import datetime
+
+from django.http import HttpResponse
+from django.utils.translation import ugettext as _
+from django.views.decorators.http import require_POST
+
+from .api import perform_search
+
+# log appears to be standard name used for logger
+log = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+class DateTimeEncoder(json.JSONEncoder):
+
+    """ encode datetimes into json appropriately """
+
+    def default(self, obj):  # pylint: disable=method-hidden
+        """ override default encoding """
+        if isinstance(obj, datetime.datetime):
+            encoded_object = obj.isoformat()
+        else:
+            encoded_object = super(DateTimeEncoder, self).default(self, obj)
+        return encoded_object
+
+
+@require_POST
+def do_search(request, course_id=None):
+    """
+    Search view for http requests
+    """
```

Comment:
Anything in edxseach subfigure is for testing. This implements the real url handler, and the application including it optionally overrides SearchResultProcessor and SearchFilterGenerator to blend in custom filtering / post-processing.

Yes, comment herein can be added.